### PR TITLE
202002 New Style

### LINKED ISSUE #66: AIRAC 2020/02 (The New Way)
**New Aerodrome Positions**

- EGBB_TWR changes frequency to 123.97
- EGMC_TWR changes frequency to 127.75
- Add EGLF_GND on 121.820

**TC North East Splits**
- No prenote changes
- No topdown changes
- Added TC Lambourne (LTC_NL_CTR, 123.900)
- Added TC Lorel (LTC_NR_CTR, 129.720)

**Clacton Splits**
- Added LON_EN_CTR (133.950)
- Added LON_ES_CTR (128.150)
- No top-down changes

**Daventry Splits**
LON_CE_CTR - Daventy East - 127.870
LON_CW_CTR - Daventry West - 134.400
LON_CL_CTR - Daventry Low - 133.970
LTC_M_CTR - TC Midlands - 121.020
LTC_MW_CTR - TC WELIN - 130.920
LTC_MC_CTR - TC COWLY - 133.070

Top down changes:

EGBB/EGTK
BB TWR
BB RAD
LTC_MC_CTR
LTC_M_CTR
LON_CW_CTR
LON_C_CTR

EGNX/EGTC
NX TWR
NX RAD
LTC_MW_CTR
LTC_M_CTR
LON_CL_CTR
LON_CE_CTR
LON_C_CTR

EGSS/EGGW/EGSC
...
...
LTC_CTR
LON_CL_CTR
LON_CE_CTR
LON_C_CTR

### PR DESCRIPTION
**Pull request by AndyTWF**
Close #66 

- Also adds the handoff dependencies.

### CODE REVIEW

File: app/Services/AirfieldService.php [hunk 1]
```diff
@@ -13,23 +16,168 @@ public function getAllAirfieldsWithRelations() : array
...
+            $controllersToChange->each(function (ControllerPosition $position) use ($airfield) {
+                DB::table('top_downs')
+                    ->where(['controller_position_id' => $position->id, 'airfield_id' => $airfield->id])
+                    ->update(['order' => DB::raw('`order` + 1')]);
+            });
+
+            // Pop the new position in
+            $airfield->controllers()->attach($insertPosition->id, ['order' => $airfieldBefore->pivot->order]);
+            DB::commit();
+        } catch (Exception $exception) {
```
Comment (AxonC) [inline app/Services/AirfieldService.php]: I probably know the answer, but is there a more specific exception we can catch here?

Comment (AxonC) [reply app/Services/AirfieldService.php]: I guess the behaviour would be the same regardless of the exception flow, whether it be a model not found etc.

File: database/migrations/2020_01_30_202040_airac_202002_daventry_splits.php [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+            'frequency' => 127.87,
+            'top_down' => [
+                'EGNX' => 'LON_C_CTR',
+                'EGTC' => 'LON_C_CTR',
+                'EGSS' => 'LON_C_CTR',
+                'EGGW' => 'LON_C_CTR',
+                'EGSC' => 'LON_C_CTR',
+            ],
+            'handoff' => [
+                'EGNX_SID_NORTH_09' => 'LON_C_CTR',
```
Comment (AxonC) [inline database/migrations/2020_01_30_202040_airac_202002_daventry_splits.php]: Is this right? Assuming this is LON_C_CTR in absence of NX APC?

File: app/Services/AirfieldService.php (hunk 1 above)
Comment (AndyTWF) [reply app/Services/AirfieldService.php]: Logic here is that if anything at all has gone wrong, we probably want to kill the migration so we don't leave the data half-cocked :)

File: database/migrations/2020_01_30_202040_airac_202002_daventry_splits.php (hunk 2 above)
Comment (AndyTWF) [reply database/migrations/2020_01_30_202040_airac_202002_daventry_splits.php]: Sorry - what it's doing is it's effectively saying:

"Insert this position into the handoff order `EGNX_SID_NORTH_09` before `LON_C_CTR`"

File: app/Services/AirfieldService.php (hunk 1 above)
Comment (AxonC) [reply app/Services/AirfieldService.php]: All good!

File: database/migrations/2020_01_30_202040_airac_202002_daventry_splits.php (hunk 2 above)
Comment (AxonC) [reply database/migrations/2020_01_30_202040_airac_202002_daventry_splits.php]: Oh got it - all good!

**Pull request merged**